inc/vcf/fixer.hpp

```diff
@@ -108,9 +109,48 @@ namespace ebi
             util::writeline(output, *line);
             ignored_errors++;
         }
+        /**
+         * fix: remove the info field that caused the error
+         */
         virtual void visit(InfoBodyError &error) override
         {
-            ignored_errors++;
+            // TODO better log system, if any
+            std::cerr << "DEBUG: line " << error.get_line() << ": fixing invalid INFO field " << error.get_field()
+                      << std::endl;
+            int fixed = 0;
+
+            std::vector<std::string> columns;
+            std::string string_line{line->begin(), line->end()};
+            util::string_split(string_line, "\t", columns);
+            std::vector<std::string> infos;
+            util::string_split(columns[7], ";", infos);
+
+            for (size_t i = 0; i < 7; ++i) {
+                output << columns[i] << "\t";
+            }
+            std::vector<std::string> key_value;
+            for (size_t j = 0; j < infos.size(); ++j) {
+                util::string_split(infos[j], "=", key_value);
+                if (key_value[0] != error.get_field()) {
+                    if ((fixed - j) != 0) { // if this is not the first field we don't remove, then write the separator
+                        output << ";";
+                    }
+                    output << infos[j];
+                } else {
+                    fixed++;
+                }
+            }
+
+            if (fixed == infos.size()) {
+                output << ".";
+            }
+            for (size_t i = 8; i < columns.size(); ++i) {
+                output << "\t" << columns[i];
+            }
```

Comment:
By using a column number as suggested [here](https://github.com/EBIvariation/vcf-validator/pull/43/files#r76741086), an auxiliary method `write_correct_fields` or similar could be extracted.